analysis.py

```diff
@@ -431,22 +431,27 @@ def calculate_indirect_impacts_from_df(df, io_a, config, direct_output_dir, indi
         try:
             LOGGER.info(f"Calculating indirect {io_a} impacts for {logging_dict}...")
             imp = Impact.from_hdf5(row['yearset_path'])
+
+            # Strip the crop type from the sector name, as they are not in the sector list of the MRIO table
+            impacted_sector = "agriculture" if "agriculture" in row['sector'] else row['sector']
```

Comment:
I'd like to think carefully about how we implement shocks to agriculture now that it's split into four crops rather than one.

My understanding of this implementation is that it runs the supply chain for each of the four crop types (when requested in the config). For each of these four runs, the _entire_ agricultural sector will be shocked by the % loss to the _individual_ crop (since the model reads the impact for that crop, normalises by the crop's total value, and uses that to shock the agriculture sector).

That worked ok when we only had one crop, but calculating four different shocks to the entire sector doesn't make intuituve sense to me, and I'm not sure how we'd combine them after the supply chain is run.

I'd suggest aggregating the production loss impacts before the supply chain is run. I think this would be possible: we can sum them up across all crops and normalise them by the total crop exposure. It might require a bit of thought about how to do it elegantly in the pipeline. It'll also reduce the number of times we need to run the supply chain module when we run the pipeline.

Let me know your thoughts!

Reply:
Hi Chris, 

Yes, I agree with your point about using the agriculture supply chain pipeline multiple times for the different corps is probably not meaningful and highly overestimates the indirect losses. 

I will post this comment to the card of the agriculture aggregation

Reply:
True, thanks for pointing that out, I seem to not have though this through well enough. 
Not really sure if I understood your solution fully, I see two solutions, one includes normalization: 
1. ** Aggregate the direct impacts prior to the MRIO** In that case we would theoretically, not need to normalize by the total crop exposure. (Assuming we say that the available crop types in climada are the total agriculture sector, which is of course a strange assumption in the first place, because there are probably a lot of other agricultural goods)
2. ** Aggregate after the MRIO table ** In that case we would have compute a weight for each individual country and crop to the whole countries agriculture exposure. E.g. if rice contributes 30% to the whole agriculture exposure of the US, we would have to multiple the direct impact values with 0.3. 

But maybe I'm missing something @ChrisFairless ?

Reply:
Good thoughts – and you're right, my suggestion wasn't fully thought-through.

I like your first suggestion here. It makes more intuitive sense to me to calculate the direct shocks and _then_ run the MRIO, rather than the other way around (even if the results are identical). I guess it'll be faster as well, since we'll have 1/4 of the agricultural supply chains to run.

We'll need to make sure we get the aggregation right: I guess the calculation for the total fractional impact is

$$
\frac{\displaystyle\sum_{crops} impact_{crop}}{\displaystyle\sum_{crops} exposure_{crop}}
$$

Is that right?

(TIL you can write formulae in GirHub comments. It's a good day)

Sorry for the slow response!